<!DOCTYPE html>
<html>
  <head>
    <title>
      Blog
    </title>
    <meta charset="utf-8">
    <meta name="description" content="">
    <meta name="language" content="en">
    <meta name="viewport" content="width=device-width, initial-scale=1">

    <link rel="stylesheet" href="../../static/style.css">

    <link rel="alternate" type="application/rss+xml" title="RSS feed for the blog" href="/rss.xml">

    <!--google-->
    <script async src="https://www.googletagmanager.com/gtag/js?id=G-MTNZ0ZSG3W"></script>
    <script>
      window.dataLayer = window.dataLayer || [];
      function gtag(){dataLayer.push(arguments);}
      gtag('js', new Date());
      gtag('config', 'G-MTNZ0ZSG3W');
    </script>

  </head>
  <body>
    <ul class="menu-list">
      <li class="menu-item"><a href="index.html" class="menu-link menu-title">Corin Wagen</a></li>
      <li class="menu-item"><a href="index.html#about" class="menu-link">About</a></li>
      <li class="menu-item"><a href="index.html#projects" class="menu-link">Projects</a></li>
      <!--<li class="menu-item"><a href="index.html#past_work" class="menu-link">Past Work</a></li>-->
      <li class="menu-item">
        <a href="blog_p1.html" class="menu-link">Blog</a>
        <a href='archive.html' class="menu-link">(Archive)</a>
      </li>
    </ul>
    <h1 class='blogroll-header'>Blog</h1><div class='previous-link'><a href='blog_p7.html'>previous page</a></div><div class='next-link'><a href='blog_p9.html'>next page</a></div><br><div class='blogroll-container'><h2><a class='blogroll-title' href='../../public/blog/20230215_science_engineering.html'>Science and Engineering</a></h2><i>February 15, 2023</i>
<blockquote><i>
In science, if you know what you are doing, you should not be doing it. In engineering, if you do not know what you are doing, you should not be doing it.
</i></blockquote>

<p style="text-align:right;">
—Richard Hamming
</p>

<p>
What’s the difference between science and engineering? 
</p>

<p>
Five years ago, I would have said something along the lines of “engineers study known unknowns, scientists study unknown unknowns” (with apologies to <a href=https://en.wikipedia.org/wiki/There_are_unknown_unknowns>Donald Rumsfeld</a>), or made a distinction between expanding the frontiers of knowledge (science) and settling already-explored territory (engineering).
</p>

<p>
These thoughts seem broadly consistent with what others think. Boston University’s <a href=https://www.bu.edu/eng/about-eng/meet-the-dean/engineering-is-not-science/>College of Engineering</a> says:
</p>

<blockquote>
Engineers are not a sub-category of scientists. So often the two terms are used interchangeably, but they are separate, albeit related, disciplines. Scientists explore the natural world and show us how and why it is as it is. Discovery is the essence of science. Engineers innovate solutions to real-world challenges in society. While it is true that engineering without science could be haphazard; without engineering, scientific discovery would be a merely an academic pursuit.
</blockquote>

<p>
And the <a href=https://www.nspe.org/resources/press-room/resources/frequently-asked-questions-about-engineering>National Society of Professional Engineers</a> says:
<p>

<blockquote>
Science is knowledge based on observed facts and tested truths arranged in an orderly system that can be validated and communicated to other people. Engineering is the creative application of scientific principles used to plan, build, direct, guide, manage, or work on systems to maintain and improve our daily lives.
</blockquote>

<p>
As I’ve started thinking more about the structure of the modern research system, and what its proper scope and purpose should be, I’ve grown increasingly skeptical of these distinctions. The claim I want to make in this post is that, following the above definitions of engineering, <u>most chemistry is engineering</u>. I don’t think this is bad! In fact, I think that many chemists could benefit from borrowing from an engineering mindset, and should consider incorporating this perspective in their self-conception.
<p>

<h2>
Much of Organic Chemistry is Engineering
</h2>

<p>
I want to return to the BU and NSPE definitions, because I think they’re concise and well-written, and take as gospel that scientists “explore the natural world and show us how and why it is as it is,” while engineers “innovate solutions to real-world challenges in society” (we’ll revisit issues of definition later). In short, developing something you want other people to use makes you an engineer. Which branches of modern organic chemistry are science, and which are engineering? 
<p>

<p>
Method development—one of my core interests—seems like a good candidate for “engineering.” Researchers in this space identify unsolved problems in organic synthesis, develop methods or catalysts to solve these problems, and then (in many cases) advertise, license, &amp; sell their solutions to consumers! (If you don’t believe me, just look at the <a href=https://www.sigmaaldrich.com/US/en/collections/professor-product-portal>“Organic Synthesis” tab</a> of the Sigma-Aldrich Professor Product Portal.) If these products weren’t molecules and were instead mechanical gadgets, nothing about this would be obviously scientific.
<p>

<p>
And the problems chosen are identified almost purely on the basis of what might be useful to potential users. There’s no clearer illustration of this than the recent gold rush to identify synthetic routes to bicyclic arene bioisosteres, which are useful in medicinal chemistry. Five years ago, I can’t think of a single paper making these compounds; now, I can find nine in high-profile journals just from the past year or so (<a href=https://www.nature.com/articles/s41586-022-05290-z.pdf>1</a>,
<a href=https://www.nature.com/articles/s41557-022-00979-0.pdf>2</a>,
<a href=https://pubs.acs.org/doi/pdf/10.1021/jacs.2c09733>3</a>,
<a href=https://pubs.acs.org/doi/pdf/10.1021/jacs.2c11501>4</a>,
<a href=https://pubs.acs.org/doi/pdf/10.1021/acscatal.2c03498>5</a>,
<a href=https://pubs.acs.org/doi/pdf/10.1021/acs.orglett.1c03681>6</a>,
<a href=https://pubs.acs.org/doi/pdf/10.1021/acs.orglett.2c03606>7</a>,
<a href=https://onlinelibrary.wiley.com/doi/pdfdirect/10.1002/anie.202205103>8</a>,
<a href=https://www.nature.com/articles/s41557-023-01135-y>9</a>).
<p>

<p>
Mechanistic and physical organic chemistry—another love of mine—present a tougher case, since in most cases the outcome of the studies is only knowledge. But I’m still not sure that this makes this field scientific! Let me illustrate why with an example. 
<p>

<p>
An automotive engineer may be confused by why a given transmission is not working. He/she may formulate a hypothesis, take the transmission apart, perform various experiments on its constituent pieces, validate or disprove the initial hypothesis, and generally conduct a thorough mechanistic investigation to understand the origin of the problem. But does that make him/her a scientist?
<p>

<p>
The answer, I think, is no. The subject matter is not scientific, so no amount of scientific thinking can make the work science. Similarly, I’d argue that investigating the mechanism of a system invented and developed by humans—like a Pd-catalyzed cross-coupling reaction—doesn’t count as science. (Does application of the scientific method automatically make one a scientist? See below for a continued discussion.)
<p>

<p>
In contrast, something that I think is a truly scientific area of investigation is the study of enzyme structure and function. Despite extensive study and many Nobel prizes, we’re still learning about how enzymes operate and how they achieve such extraordinary reactivity and selectivity. (For an example of this sort of work, see Arieh Warshel’s <a href=https://pubs.acs.org/doi/full/10.1021/cr0503106>review</a> on electrostatic effects in enzyme catalysis, and references therein.)
<p>

<p>
I don’t think I understand all areas of chemistry well enough to fairly judge whether they’re better understood as engineering or science, so I’ll leave this as an exercise to the reader: What motivates your research? Are you mainly driven by a desire to understand the natural order of things, or do you aim to develop technology to make the world better? Both are important, and neither answer is bad—but if your main goal is inventing a new molecule, material, algorithm, or medicine, you might consider thinking of yourself as more of an engineer than a scientist.
<p>

<h2>
Do Different Definitions Clarify Matters?
</h2>

<p>
Since the claim that “most chemistry is engineering” is weird, we might consider alternative definitions to solve this problem.
<p>

<p>
One appealing definition: “a scientist is anyone who uses the scientific method.” As I discussed above, in the case of the automotive engineer, lots of people use the scientific method who clearly aren’t scientists: engineers, yes, but also detectives, doctors, and many other people. Indeed, according to this definition almost anyone who acquires data to shed light on a problem is “doing science.” So I don’t think this is a very good definition.
<p>

<p>
Another definition might be: “if you’re closely involved with science, you’re a scientist, even if the work you’re doing isn’t literally pushing the frontiers of knowledge forward.” I’m sympathetic to this definition, but I still find it hard to separate scientists and engineers here. What makes an engineer optimizing a new flow reactor less of a scientist than the chemist optimizing a new catalyst? Are they both scientists? What about people who work on chip design and fabrication, or people who design analytical instruments, or people who use them? I can’t find any clean way to divide scientists from engineers that recapitulates the conventional usage of the terms.
<p>

<h2>
Why Does This Matter?
</h2>

<p>
I think the root of this confusion is that <u>the nature of scientific fields has changed over the past half-century</u>. Organic chemistry hasn’t always been largely engineering; a century ago, the structure of natural products and the nature of the chemical bond were mysteries, truly the domain of science, and these issues were studied by chemists. As we’ve grown to understand our field better and better, our work has shifted from science to engineering—the true mysteries in chemistry are now few and far between, and the challenge facing today’s chemists is how to use centuries of accumulated knowledge to better society. But because of our lineage, we think of ourselves as scientists, and have managed to disguise the true nature of our work so well that we’ve deceived even ourselves.
</p>

<p>
By this point, it should be obvious that I don’t think science is superior to engineering. In fact, I’m glad to work in an area that’s largely engineering! But the way that an engineer ought to approach their work is different from the way a scientist ought to approach their work. In writing this piece, I came across a 1996 <a href=https://dl.acm.org/doi/10.1145/227234.227243>article</a> by Frederick Brooks, who argued that computer science was better understood as an engineering discipline and defended the importance of this claim: 
</p>

<blockquote>
If our discipline has been misnamed, so what? Surely computer science is a harmless conceit. What’s in a name? Much. Our self-misnaming hastens various unhappy trends.
<br><br>
First, it implies that we accept a perceived pecking order that respects natural scientists highly and engineers less so, and that we seek to appropriate the higher station for ourselves. That is a self-serving gambit, hence dubious….
<br><br>
Second, sciences legitimately take the discovery of facts and laws as a proper end in itself. A new fact, a new law is an accomplishment, worthy of publication…. But in design, in contrast with science, novelty in itself has no merit. If we recognize our artifacts as tools, we test them by their usefulness and their costs, not their novelty.
<br><br>
Third, we tend to forget our users and their real problems, climbing into our ivory towers to dissect tractable abstractions of those problems, abstractions that may have left behind the essence of the real problem. 
</blockquote>

<p>
I think Brooks’s summary is simple and elegant. If we judge the value of our work based on the utility of our tools, rather than the novelty of our ideas, we’ll spend our time on different problems and get excited about different advances. There’s room for both scientists and engineers in chemistry—but at the margin, I think our field would benefit from becoming less like science, and more like engineering.
</p>

<i>
Thanks to Michael Nielsen, Joe Gair, Ari Wagen, and Michael Tartre for editing drafts of this post. Michael Tartre sent me the Richard Hamming quote, and Phil Brooks sent me his grandfather’s article.
</i>

</div><div class='blogroll-container'><h2><a class='blogroll-title' href='../../public/blog/20230203_poy.html'>2022 Paper(s) of the Year</a></h2><i>February 3, 2023</i>
<p>
Every year, our group participates in a “Paper of the Year” competition, where we each nominate five papers and then duke it out in a multi-hour debate. Looking through hundreds of papers in a few weeks is a great exercise: it helps highlight both creativity and its absence, and points towards where the field’s focus might turn next.
</p>

<p>
My picks are very personal—how could they not be?—and also biased towards towards experimental chemistry, owing to what our group focuses on. So, don’t take this as any attempt towards creating an objective list.
</p>

<p>
All the papers I really liked are listed below, with my top five listed in bold:
</p>

<ul>
<li><a href=https://www.science.org/doi/pdf/10.1126/science.abo0039>A lovely electrochemical cross-electrophile coupling</a> (Sevov). Great scope and forms challenging bonds.</li>

<li><a href=https://www.science.org/doi/pdf/10.1126/science.abo4282>Carbon deletion through photochemistry, converting quinolines to indoles</a> (Levin). 
My favorite of the skeletal editing papers, with apologies to the others (<a href=https://pubs.acs.org/doi/pdf/10.1021/jacs.2c09616>1</a>,
<a href=https://pubs.acs.org/doi/pdf/10.1021/jacs.2c10746>2</a>, 
<a href=https://www.science.org/doi/10.1126/science.add1383>3</a>,
<a href=https://pubs.acs.org/doi/pdf/10.1021/jacs.2c10570>4</a>,
<a href=https://onlinelibrary.wiley.com/doi/pdfdirect/10.1002/anie.202202703>5</a>, 
<a href=https://onlinelibrary.wiley.com/doi/pdfdirect/10.1002/anie.202208014>6</a>,
<a href=https://pubs.acs.org/doi/10.1021/jacs.2c08464>7</a>). <b>Pick 1/5.</b></li>

<li><a href=https://www.science.org/doi/pdf/10.1126/science.abn1885>High-throughput additive screening reveals a new role for phthalimide in Ni-catalyzed cross-couplings</a> (MacMillan, Dreher).</li>

<li><a href=https://www.science.org/doi/10.1126/science.abo6443>Turning aldehydes into metal carbenoids</a> (Nagib); self-recommending.</li>

<li><a href=https://www.science.org/doi/10.1126/science.add6852>Epimerization at tertiary carbons for “stereochemical editing”</a> (Wendlandt).</li>

<li><a href=https://www.science.org/doi/10.1126/science.ade5320>Setting stereocenters with vinyl carbocations, somehow</a> (Nelson, Houk).</li>

<li><a href=https://www.nature.com/articles/s41586-022-04491-w.pdf>Expanding automated synthesis to Csp<sub>3</sub> centers with “TIDA boronates”</a> (Burke). 
An approach with a real chance of revolutionizing how we approach organic synthesis. <b>Pick 2/5.</b></li>

<li><a href=https://www.nature.com/articles/s41586-022-04524-4.pdf>Some beautiful asymmetric catalysis to form S-chiral sulfinate esters</a> (Tan).</li>

<li><a href=https://www.nature.com/articles/s41586-022-05211-0.pdf>Two</a>
<a href=https://pubs.acs.org/doi/pdf/10.1021/jacs.2c05648>papers</a> disclosing the use of photoexcited nitroarenes to perform “ozonolysis without ozone” (Parasram, Leonori).

<li><a href=https://www.nature.com/articles/s41557-022-00895-3.pdf>Enantioselective conversion of carboxylic acids into amino acids</a> (Meggers, Chen).</li>

<li><a href=https://pubs.acs.org/doi/pdf/10.1021/jacs.2c00527>Catalytic olefin hydroalkoxylation, through a tri-catalytic system</a> (Ohmiya). Reminds me a little of <a href=https://pubs.acs.org/doi/full/10.1021/jacs.0c04735>Kanai’s tri-catalytic aldehyde allylation</a>; in both cases, it’s hard to believe so many catalysts play nicely together.</li>

<li><a href=https://pubs.acs.org/doi/10.1021/jacs.2c04807>Deoxygenative trifluoromethylation</a> (MacMillan), a reaction which speaks for itself.</li>

<li><a href=https://pubs.acs.org/doi/pdf/10.1021/jacs.2c08778>Functional group-tolerant Kochi–Salamon [2+2] cycloaddition in water, catalyzed by copper sulfate</a> (Burns). Bizarre, unbelievable, outstanding. <b>Pick 3/5.</b> (With apologies to the other 2+2 cycloaddition papers:
<a href=https://www.nature.com/articles/s41586-022-05335-3.pdf>1</a>,
<a href=https://www.nature.com/articles/s41586-022-05342-4.pdf>2</a>,
<a href=https://www.nature.com/articles/s41586-022-04636-x.pdf>3</a>,
<a href=https://pubs.acs.org/doi/pdf/10.1021/acs.orglett.2c03606>4</a>,
<a href=https://onlinelibrary.wiley.com/doi/pdfdirect/10.1002/anie.202200725>5</a>,
<a href=https://onlinelibrary.wiley.com/doi/pdfdirect/10.1002/anie.202208800>6</a>,
<a href=https://onlinelibrary.wiley.com/doi/pdfdirect/10.1002/anie.202211596>7</a>.)</li>

<li><a href=https://pubs.acs.org/doi/pdf/10.1021/jacs.2c08332>Pd-catalyzed C–H hydroxylation</a> (Yu). Addresses a lot of my misgivings about C–H activation: directed by carboxylic acids, silver-free, scalable, and uses aqueous H<sub>2</sub>O<sub>2</sub> (perhaps the best oxidant possible, after air).</li>

<li><a href=https://pubs.acs.org/doi/pdf/10.1021/jacs.2c09745>Red-light-mediated C–N coupling, through Os photocatalysis</a> (Rovis). Suppresses a lot of the annoying mass balance issues that plague conventional photoredox.</li>

<li><a href=https://pubs.acs.org/doi/10.1021/jacs.1c13151>A nice mechanistic study of an enantioselective C–N coupling reaction</a> (Peters, Fu).</li>

<li><a href=https://pubs.acs.org/doi/pdf/10.1021/jacs.2c01487>Mechanistic study and an improved phosphetane catalyst for C–N coupling</a> (Radosevich).</li>

<li><a href=https://pubs.acs.org/doi/pdf/10.1021/jacs.2c01375>Studying the factors that lead to hydroxyl versus chlorine radical transfer in non-heme Fe complexes</a> (Goldberg, de Visser).</li>

<li><a href=https://pubs.acs.org/doi/pdf/10.1021/jacs.2c02283>Studying the mechanism of enantioselective enzymatic C–H amination</a> (Yang Yang, Peng Liu). Surprisingly, radical rebound is enantiodetermining! I remember being puzzled by the mechanism of this reaction when it was first published; this is a very satisfying resolution. Cool isotope studies &amp; <i>ab initio</i> molecular dynamics seal the deal. <b>Pick 4/5.</b></li>

<li><a href=https://pubs.acs.org/doi/pdf/10.1021/jacs.2c01356>The latest on photochemistry of Ni oxidative-addition complexes</a> (Hadt). Beautiful physical inorganic chemistry; a lot of the details are beyond me, but clearly important work.</li>

<li><a href=https://pubs.acs.org/doi/10.1021/jacs.2c07099>Mechanistic study of PCET-mediated olefin hydroamination</a> (Knowles).</li>

<li><a href=https://pubs.acs.org/doi/10.1021/jacs.2c07643>A really beautiful metal-free hydroamination</a> (Wang/Wang).</li>

<li><a href=https://pubs.acs.org/doi/10.1021/jacs.2c09786>The latest on cobaltacene-based PCET mediators</a> (Peters). Big fan of this program.</li>

<li><a href=https://pubs.acs.org/doi/pdf/10.1021/jacs.2c00675>Using aryl nitriles as sensors for electric field strength</a> (Boxer). 
Also a really cool program: I liked <a href=https://chemrxiv.org/engage/chemrxiv/article-details/626b29f9ed4d8830581c6ed6>this preprint</a> too.
</li>

<li><a href=https://pubs.acs.org/doi/10.1021/jacs.2c00154>Studying cation–acetonitrile dynamics with 2D IR and isotope labelling</a> (Tokmakoff). I've <a href=https://corinwagen.github.io/public/blog/20220708_apotheosis_of_2d_ir.html>written about this</a> before</li>

<li><a href=https://pubs.acs.org/doi/10.1021/acscatal.1c05802>A very nice <sup>13</sup>C KIE study of the Suzuki reaction, emphasizing the importance of monoligated Pd even when using PPh<sub>3</sub></a> (Vetticatt, Hirschi). And another <a href=https://pubs.acs.org/doi/pdf/10.1021/jacs.2c02941>cool <sup>13</sup>C KIE study</a> for Hirschi.</li>

<li><a href=https://onlinelibrary.wiley.com/doi/pdfdirect/10.1002/anie.202114044>The first enantioselective beta-alkoxy elimination I’ve seen!</a> (Streuff)</li>

<li><a href=https://pubs.acs.org/doi/pdf/10.1021/acs.orglett.2c03201>An interesting way to functionalize ketones, via phosphorus-mediated umpolung</a> (Ball).</li>

<li><a href=https://pubs.acs.org/doi/pdf/10.1021/acs.organomet.2c00504>Investigating a concentration-dependent KIE for protonolysis of (cod)PtMe<sub>2</sub></a> (Bowring).</li>

<li><a href=https://onlinelibrary.wiley.com/doi/pdfdirect/10.1002/anie.202213056>Biocatalytic N-heterocycle methylation, with unbelievable regioselectivity</a> (Hammer). Hammer also published a <a href=https://onlinelibrary.wiley.com/doi/10.1002/anie.202215093>really exciting styrene hydration paper</a>, albeit technically in 2023.</li>

<li><a href=https://pubs.acs.org/doi/10.1021/acscatal.2c04316>Bifunctional redox-active esters add to styrenes to make a variety of heterocycles.</a> (Knowles/Doyle).</li>

<li><a href=https://pubs.rsc.org/en/content/articlehtml/2022/sc/d2sc02274e>A nice illustration of the importance of conformer sampling workflows</a> (Neese, Bistoni). See also work from 
<a href=https://pubs.rsc.org/en/content/articlepdf/2022/sc/d2sc01714h>Laplaza/Corminboef</a>, 
<a href=https://pubs.acs.org/doi/pdf/10.1021/jacs.2c01381>Peter Chen</a>, and <a href=https://onlinelibrary.wiley.com/doi/pdfdirect/10.1002/anie.202113905>Grimme</a>.</li>

<li><a href=https://arxiv.org/pdf/2204.05249.pdf>The latest equivariant neural network potential</a>, faster than <a href=https://www.nature.com/articles/s41467-022-29939-5>their previous work</a> (Kozinsky).</li>

<li><a href=https://arxiv.org/pdf/2010.01196.pdf>Espaloma, a differential neural network forcefield from the OpenForcefield folks.</a></li>

<li><a href=https://chemrxiv.org/engage/chemrxiv/article-details/62b1b0c97da6ce535c19d40c>Studying cycloaddition dynamics with neural network potentials</a> (Young, Duarte).</li>

<li><a href=https://pubs.acs.org/doi/pdf/10.1021/acs.jpca.2c08301>Studying dynamic control of kinetic product ratios in cyclopropylidine opening using ML-learned potentials</a> (Carpenter).</li>

<li><a href=https://aip.scitation.org/doi/full/10.1063/5.0133026>wB97X-3c, the latest “composite method”</a> (Grimme). If the reported speed and accuracy are true “in the wild,” every computational chemist ought to use this method. <b>Pick 5/5.</b></li>
</ul>
</div><div class='blogroll-container'><h2><a class='blogroll-title' href='../../public/blog/20230123_integral_screening.html'>The Importance of Integral Screening</a></h2><i>January 23, 2023</i>
<p>
For almost all Hartree–Fock-based computational methods, including density-functional theory, the rate-limiting step is calculating electron–electron repulsion. (This isn’t true for semiempirical methods, where matrix diagonalization is generally rate-limiting, or for <a href=https://pubs.acs.org/doi/10.1021/acs.jctc.2c00876>calculations on very large systems</a>.)
</p>

<p>
When isolated single molecules are the subject of calculations (as opposed to solids or periodic systems), most programs describe electronic structure in terms of atom-centered basis sets, which reduces the electron–electron repulsion problem to one of calculating electron repulsion integrals (ERIs) over quartets of basis shells. Framed this way, it becomes obvious why ERIs are the bottleneck: the number of ERIs will scale as O(N<sup>4</sup>), meaning that millions of these integrals must be calculated even for relatively small molecules.
</p>

<figure>
  <img class="centered-img" src="../img/20230125_dalle.png" style="width:350px;" />
  <figcaption> 
  I tried to get DALL-E to make a visual representation of integral screening; this was the best I got.
  </figcaption>
</figure>

<p>
One big advance in electronic structure calculations was the development of integral screening techniques, the most popular of which is the “Schwartz inequality” (derived from the <a href=https://en.wikipedia.org/wiki/Cauchy%E2%80%93Schwarz_inequality>Cauchy–Schwartz inequality</a>, but actually developed by <a href=https://onlinelibrary.wiley.com/doi/10.1002/jcc.540100111>Mario Häser and Reinhart Ahlrichs</a> [<b>EDIT: This is wrong, see correction at end!</b>]). If we denote an ERI over shells A, B, C, and D as (AB|CD), then the Schwartz inequality says:
</p>

<p>
(AB|CD) ≤ (AB|AB)<sup>0.5</sup> (CD|CD)<sup>0.5</sup>
</p>

<p>
This is pretty intuitive: each shell pair will interact with itself most, since it has perfect overlap with itself, and so the geometric mean of the interaction of each shell pair with itself is an upper bound for the interaction between the two shell pairs. (Why would (AB|AB) ever be a small value? Well, A and B might be super far away from each other, and so the “shell pair” has very little overlap is just negligible.)
</p>

<p>
This result is very useful. Since there are many fewer integrals of the form (AB|AB), we can start by calculating all of those, and then use the resulting values to “screen” each shell quartet. If the predicted value is less than some predefined cutoff, the integral is skipped. While these screening methods don’t help much with small molecules, where all of the shells are pretty close to each other, they become crucial for medium-sized molecules and above.
</p>

<p>
(What’s the cutoff value? Orca defaults to 10<sup>-8</sup>, <a href=https://gaussian.com/integral/>Gaussian</a> to 10<sup>-12</sup>, <a href=https://psicode.org/psi4manual/master/autodir_options_c/module__scf.html>Psi4</a> to 10<sup>-12</sup>, and <a href=https://manual.q-chem.com/5.2/Ch4.S3.SS2.html>QChem</a> to 10<sup>-8</sup>–10<sup>-10</sup> depending on the type of calculation.)
</p>

<p>
The Schwartz inequality neglects, however, another way in which (AB|CD) might be very small: if (AB| and |CD) aren’t independently negligible, but are just really far away from each other. One elegant way to address this (out of many) comes from recent-ish work by <a href=https://aip.scitation.org/doi/10.1063/1.4994190>Travis Thompson and Christian Ochsenfeld</a>. They define an intermediate quantity M for each pair of shells, derived from different high-symmetry integrals:
</p>

<p>
M<sub>AC</sub> := (AA|CC) /  ( (AA|AA)<sup>0.5</sup> (CC|CC)<sup>0.5</sup> )
</p>

<p>
M<sub>AC</sub> intuitively represents the distance between the two shells, and is guaranteed to be in the range [0,1]. Thompson and Ochsenfeld then use this quantity to propose an estimate of a shell quartet’s value:
</p>

<p>
(AB|CD) ≈ (AB|AB)<sup>0.5</sup> (CD|CD)<sup>0.5</sup> max(M<sub>AC</sub>M<sub>BD</sub>, M<sub>AD</sub>M<sub>BC</sub>)
</p>

<p>
This is no longer a rigorous upper bound like the Schwartz inequality, but it’s a pretty good estimate of the size of the integral.
</p>

How much of a difference does this make in practice? To test this, I ran HF/STO-3G calculations on dodecane in the fully linear configuration. As shown by <a href=https://onlinelibrary.wiley.com/doi/10.1002/jcc.540030314>Almlöf, Faegri, and Korsell</a>, linear molecules benefit the most from integral screening (since the shells are on average farther apart), so I hoped to see a sizable effect without having to study particularly large molecules.
</p>

<figure>
  <img class="centered-img" src="../img/20230125_graph.png" style="width:450px;" />
  <figcaption> 
  Almlöf, Faegri, and Korsell, Figure 5. This paper is terrific.
  </figcaption>
</figure>

<p>
I compared both the Schwartz (“QQ”) bound and Ochsenfeld’s CSAM bound for integral thresholds ranging from 10<sup>-9</sup> to 10<sup>-13</sup>, and compared the result to a calculation without any integral screening. The total time for the calculation, as a percent of the unscreened time, is plotted below against the error in µHartree (for the organic chemists out there, 1 µH = 0.00063 kcal/mol):
</p>

<figure>
  <img class="centered-img" src="../img/20230125_graph2.png" style="width:400px;" />
  <figcaption> 
  Comparing CSAM and QQ.
  </figcaption>
</figure>

<p>
A few things are apparent for this data. First, even tight thresholds lead to dramatic speedups relative to the unscreened calculation—and with minimal errors. Secondly, the CSAM bound really does work better than the QQ bound (especially if you ignore the high-error 10<sup>-9</sup> threshold data point). For most threshold values, using CSAM leads to about a 20% increase in speed, at the cost of a 3-fold increase in an already small error. Viewed visually, we can see that the Pareto frontier for CSAM (blue) is just closer to the optimal bottom-left corner than the corresponding frontier for QQ (black).
</p>

<p>
I hope this post serves to explain some of the magic that goes on behind the scenes to make “routine” QM calculations possible. (If you thought these tricks were sneaky, wait until you hear how the integrals that aren’t screened out are calculated!)
</p>

<p>
<i>
<b>CORRECTION</b>: In this post, I credited Mario Häser and Reinhart Ahlrichs with developing the Cauchy–Schwartz method for integral screening. A (famous) theoretical chemist who shall remain nameless reached out to me to correct the record—in fact, Almlöf included an overlap-based screening method in <a href=https://onlinelibrary.wiley.com/doi/10.1002/jcc.540030314>his landmark 1982 paper</a>. To the untrained eye, this appears unrelated to ERI-based screening, but we are using Gaussian basis sets and so “one can therefore write the integrals in terms of overlaps,” meaning that what looked like a different expression is actually the same thing. (Section 9.12 of Helgaker/Jorgensen/Olsen's textbook </i>Molecular Electronic Structure Theory<i>, a book I sadly do not own, apparently discusses this more.) 
</i>
</p>

<p>
<i>
The professor traced this back to <a href=https://aip.scitation.org/doi/abs/10.1063/1.1681647?journalCode=jcp>Wilhite and Eumena</a> in 1974, and ultimately back to the work of Witten in the 1960s. It is a pleasure to get corrected by those you respect, and I welcome any readers who find errors in my writing to reach out; I will do my best to respond and take blame as appropriate.
</i>
</p>
</div><div class='blogroll-container'><h2><a class='blogroll-title' href='../../public/blog/20230118_meta_optimization.html'>Who Will Optimize The Optimizers?</a></h2><i>January 18, 2023</i>

<p>
While looking over papers from the past year, one theme in particular stood out to me: meta-optimization, or optimizing how we optimize things.
</p>

<figure>
  <img class="centered-img" src="../img/20230118_algo.png" style="width:450px;" />
  <figcaption> 
  A generic optimization (image credit to <a href="https://mpalaourg.me/project/optimization-algorithms/">Georgios Balaouras</a>).
  </figcaption>
</figure>

<p>
Meta-optimization has long been a focus of research in computer science, where new optimization algorithms can have an incredibly high impact (e.g. <a href=https://arxiv.org/pdf/1412.6980.pdf>ADAM</a>, one of the most commonly used optimizers for neural network training). More recently, the advent of directed evolution has made optimization methods a central focus of biocatalysis, since (in many cases) the efficacy of the reaction one discovers is primarily dependent on the efficacy of the optimization method used.
</p>

<p>
In contrast, it seems that meta-optimization has historically attracted less attention from “classic” organic chemists, despite the central importance of reaction optimization to so much of what we do. This post aims to show some of the ways in which this paradigm is changing, and briefly summarize some of what I consider to be the most interesting and exciting recent advances in chemical meta-optimization.
<i>(This is a big and somewhat nebulous area, and I am certainly leaving things out or not paying due homage to everyone. Sorry in advance!)</i>
</p>

<h3>Design of Experiments, and Dealing with Discrete Variables</h3>
<p>
Perhaps the best-known optimization algorithm in chemistry is “design of experiments” (DoE), which uses statistical methods to estimate the shape of a multiparameter surface and find minima or maxima more efficiently than one-factor-at-a-time screening. (DoE is a pretty broad term that gets used to describe a lot of different techniques: for more reading, <a href=https://en.wikipedia.org/wiki/Response_surface_methodology>see</a> <a href=https://en.wikipedia.org/wiki/Central_composite_design>these</a> <a href=https://en.wikipedia.org/wiki/Box%E2%80%93Behnken_design>links</a>.)
</p>

<p>
DoE has been used for a long time, <a href=https://pubs.acs.org/doi/full/10.1021/op500169m>especially in process chemistry</a>, and is very effective at optimizing continuous variables (like temperature, concentration, and equivalents). However, it’s less obvious how DoE might be extended to discrete variables. (<a href=https://pubs.acs.org/doi/pdf/10.1021/ol1020898>This 2010 report</a>, from scientists at Eli Lilly, reports the use of DoE to optimize a palladium-catalyzed pyrazole arylation, but without many details about the statistical methods used.)
</p>

<p>
<a href=https://pubs.acs.org/doi/10.1021/op300275p>A nice paper</a> from Jonathan Moseley and co-workers illustrates why discrete variables are so tricky:
</p>

<blockquote>
How does one compare such different solvents as hexane and DMSO for example? Comparing their relative polarities, which is often related to solubility, might be one way, but this may not be the most important factor for the reaction in question. Additionally, this simple trend may not be relevant in any case, given for example that chlorinated solvents have good solubilising power despite their low-to-medium polarity (as judged by their dielectric constant). On the other hand, the high polarity and solubilising power of alcohols might be compromised in the desired reaction by their protic nature, whilst the “unrelated” hexane and DMSO are both aprotic.
<br><br>
<u>In summary, replacing any one of the discrete parameters with another does not yield a different value on the same axis of a graph, as it would for a continuous parameter; instead it requires a different graph with different axes which may have no meaningful relationship to the first one whatsoever.</u> This means that every single combination of catalyst/ligand/base/solvent is essentially a different reaction for the same two starting materials to produce the same product. <i>(emphasis added)</i>
</blockquote>

<p>
The solution the authors propose is to use principal component analysis (PCA) on molecular descriptors, such as “measurable physical factors (e.g., bp, density, bond length), or calculated and theoretical ones (e.g., electron density, Hansen solubility parameters, Kamlet–Taft solvent polarity parameters),” to convert discrete parameters into continuous ones. This general approach for handling discrete variables—generation of continuous molecular descriptors, followed by use of a dimensionality reduction algorithm—is widely used today for lots of tasks (see for instance <a href=https://pubs.rsc.org/en/content/articlelanding/2016/ob/c5ob01892g>this paper</a> on DoE for solvent screening, and <a href=https://corinwagen.github.io/public/blog/20220926_plotting_diversity.html>my previous post on UMAP</a>).
</p>

<h3>Choosing Intelligent Screening Sets</h3>

<p>
With continuous descriptors for formally discrete variables in hand, a natural next step is to use this data to choose catalyst/substrates that best cover chemical space. (This can be done with several algorithms; see <a href=https://pubs.acs.org/doi/10.1021/jm9700878?ref=PDF>this paper</a> for more discussion) In 2016, this technique was popularized by the Merck <a href=https://pubs.rsc.org/en/content/articlelanding/2016/sc/c5sc04751j>“informer library”</a> approach, which generated sets of aryl boronic esters and aryl halides that could be used to fairly evaluate new reactions against complex, drug-like substrates. (See also <a href=https://pubs.acs.org/doi/full/10.1021/acs.accounts.0c00760>this recent perspective</a> on the Merck informer libraries, and <a href=https://www.pnas.org/doi/full/10.1073/pnas.1409522111>similar work</a> from Matt Sigman a few years earlier.)
</p>

<figure>
  <img class="centered-img" src="../img/20230118_merck_informer.png" style="width:600px;" />
  <figcaption> 
  Illustration of the Merck informer library (yellow) in chemical space, compared to drugs (grey) and compounds from recent literature (blue).
  </figcaption>
</figure>

<p>
While the Merck informer libraries were intended to be shared and used by lots of research groups, recently it’s become more common for individual research groups to design their own project-specific screening sets. Abbie Doyle and co-workers <a href=https://pubs.acs.org/doi/10.1021/jacs.1c12203>kicked this off</a> in 2022 by using DFT-based descriptors, UMAP dimensionality reduction, and agglomerative hierarchical clustering to generate a maximally diverse set of commercial aryl bromides. Other groups soon followed suit: <a href=https://pubs.acs.org/doi/10.1021/acscatal.2c01813>Karl Gademann</a> used this approach to study bromotetrazine cross coupling, while Marion Emmert and co-workers at Merck <a href=https://pubs.acs.org/doi/abs/10.1021/jacs.2c10557>employed similar methods</a> to investigate azole carboxylation. (I’ve also used this approach for substrate selection!)
</p>

<p>
This approach can also be used to design intelligent sets of catalysts/ligands at the outset of a screening project. Using <a href=https://pubs.acs.org/doi/10.1021/jacs.1c09718>their “kraken” dataset of phosphine properties</a>, Tobias Gensch and Matt Sigman <a href=https://pubs.acs.org/doi/10.1021/acscatal.2c01970>proposed a set of 32 commercially available ligands</a> which aims to cover as much of phosphine chemical space as possible in an initial screen. Jason Stevens and co-workers combined this idea with the substrate-selection methods from the previous paragraph to <a href=https://pubs.acs.org/doi/10.1021/acs.organomet.2c00089>perform a detailed study</a> of Ni-catalyzed borylation under many conditions, and tested a variety of ML models on the resulting dataset. (Scott Denmark and co-workers have also used a variant of this idea, called the Universal Training Set, to <a href=https://www.science.org/doi/10.1126/science.aau5631>initialize ML-driven reaction optimization</a>.)
</p>

<h3>New Optimization Algorithms</h3>

<p>
As in every area of life, ML-based approaches have been used a lot for optimization recently. This isn’t new; Klaus Jensen and Steve Buchwald <a href=https://pubs.rsc.org/en/content/articlelanding/2016/RE/C6RE00153J>used machine learning</a> to drive autonomous optimization in 2016, and <a href=https://pubs.acs.org/doi/10.1021/acscentsci.7b00492>Richard Zare</a> published a detailed methodological study in 2017. Nevertheless, as with computational substrate selection, these techniques have come into the mainstream in the past few years.
</p>

<p>
I mentioned the work of Scott Denmark on ML-driven optimization before, and his team published two more papers on this topic last year: <a href=https://pubs.acs.org/doi/10.1021/jacs.2c08820>one on atropselective biaryl iodination</a>, and <a href=https://pubs.acs.org/doi/pdf/10.1021/acs.oprd.1c00155>one on optimization of <i>Cinchona</i> alkaloid-based phase transfer catalysts</a>. In particular, the second paper (conducted in collaboration with scientists at Merck) illustrates how an ML model can be updated with new data as optimization progresses, allowing many sequential rounds of catalyst development to be conducted.
</p>

<p>
Abbie Doyle’s group has done a lot of work on using <a href=https://en.wikipedia.org/wiki/Bayesian_optimization>Bayesian optimization</a> (BO) to drive reaction optimization. <a href=https://www.nature.com/articles/s41586-021-03213-y>Their first paper</a> in this area illustrated the capacity of BO to avoid spurious local minima, and went on to validate this approach in a variety of complex problems. Even better, they compared the results of BO to chemist-guided optimization to see if computer-driven optimization could outcompete expert intuition. To quote the paper:
</p>

<blockquote>
In total, 50 expert chemists and engineers from academia and industry played the reaction optimization game (Fig. 4c). Accordingly, the Bayesian reaction optimizer also played the game 50 times (Fig. 4b), each time starting with a different random initialization. The first point of comparison between human participants and the machine learning optimizer was their raw maximum observed yield at each step during the optimization. Humans made significantly (<i>p</i> &lt; 0.05) better initial choices than random selection, on average discovering conditions that had 15% higher yield in their first batch of experiments. <u>However, even with random initialization, within three batches of five experiments the average performance of the optimizer surpassed that of the humans.</u> Notably, in contrast to human participants, Bayesian optimization achieved &gt;99% yield 100% of the time within the experimental budget. Moreover, Bayesian optimization tended to discover globally optimal conditions (CgMe-PPh, CsOPiv or CsOAc, DMAc, 0.153 M, 105 °C) within the first 50 experiments (Fig. 4b). <i>(emphasis added)</i>
</blockquote>

<figure>
  <img class="centered-img" src="../img/20230118_bayesian_optimization.png" style="width:300px;" />
  <figcaption> 
  Average yield per batch from Bayesian optimization (black) vs. humans (blue), showing how Bayesian optimization outcompetes humans despite a worse start.
  </figcaption>
</figure>

<p>
<a href=https://pubs.acs.org/doi/abs/10.1021/jacs.2c08592>Their subsequent work</a> has made their optimization app available online, and illustrated the application of this strategy to other reactions.
</p>

<p>
Closely related is <a href=https://www.science.org/doi/10.1126/science.adc8743> this work</a> from Aspuru-Guzik, Burke, and co-workers, which uses a “matrix-down” approach to choosing representative substrates for the Suzuki reaction (similar to the substrate-selection algorithms discussed previously). The selected substrates are then subjected to automated high-throughput screening guided by an uncertainty-minimizing ML model (i.e., new reactions are chosen based on the regions of chemical space that the algorithm has the least knowledge about; this is similar to, but distinct from, Bayesian optimization). This is a pretty interesting approach, and I hope they study it further in the future. (Aspuru-Guzik has done <a href=https://arxiv.org/abs/2103.03716>lots of other work</a> in this area, including some <a href=https://pubs.acs.org/doi/10.1021/acscentsci.8b00307>Bayesian optimization</a>.)
</p>

<p>
Finally, two papers this year (that I’m aware of) put forward the idea of using multi-substrate loss functions for optimization: <a href=https://www.nature.com/articles/s41586-022-05263-2>our work on screening for generality</a> and <a href=https://chemrxiv.org/engage/chemrxiv/article-details/636a84ab80c9bf01cc8d95f9>a beautiful collaboration</a> from Song Lin, Scott Miller, and Matt Sigman. These papers used “low-tech” optimization methods that are familiar to practicing organic chemists (e.g. “screen different groups at this position”), but evaluated the output of this optimization not based on the yield/enantioselectivity of a single substrate but on aggregate metrics derived from many substrates. The results that our groups were able to uncover were good, but I’m sure adding robotics and advanced ML optimization will turbocharge this concept and find new and better catalysts with truly remarkable generality.
</p>

<h3>Conclusions</h3>

<p>
Reaction optimization is a common task in organic chemistry, but one that’s commonly done without much metacognition. Instead, many researchers will screen catalysts, substrates, and conditions based on habit or convenience, without necessarily dwelling on whether their screening procedure is optimal. While this may work well enough when you only need to optimize one or two reactions in your whole graduate school career (or when acquiring each data point takes days or weeks), <i>ad hoc</i> strategies will at some point simply fail to scale.
</p>

<p>
Organic chemistry, long the realm of “small data,” is slowly but inexorably catching up with the adjacent sciences. As progress in lab automation and instrumentation makes setting up, purifying, and analyzing large numbers of reactions easier, experimental chemists will have to figure out how to choose which reactions to run and how to handle all the data from these reactions, using tools like the ones discussed above. Like it or not, data science and cheminformatics may soon become core competencies of the modern experimental chemist!
</p>

</div><div class='blogroll-container'><h2><a class='blogroll-title' href='../../public/blog/20230103_hdkie_puzzle.html'>Misleading H/D KIE Experiments</a></h2><i>January 3, 2023</i>
<p>
<i>
Note: old versions of this post lacked a discussion of S<sub>N</sub>2. I've added an appendix which remedies this.
</i>
</p>

<p>
In <a href=https://corinwagen.github.io/public/blog/20220815_rate_determining_span.html>“The Rate-Limiting Span,”</a> I discussed how thinking in terms of the span from ground state to transition state, rather than in terms of elementary steps, can help prevent conceptual errors. Today, I want to illustrate why this is important in the context of a little H/D KIE puzzle. 
</p>

<p>
Consider the following reaction, which could conceivably proceed via an S<sub>N</sub>2 mechanism (red), an S<sub>N</sub>1 mechanism (blue), or really anywhere on the continuum:
</p>

<figure>
  <img class="centered-img" src="../img/20230103_structures.png" style="width:550px;" />
  <figcaption> 
    This is a bit of a silly reaction, admittedly. The intermolecular version or the <i>endo</i> version would have been better.
  </figcaption>
</figure>

<p>
What experiment can be used to investigate the mechanism of this reaction? One possibility is an alpha H/D KIE experiment at the iminium position. Reactions with a sp<sup>3</sup> ground state and an sp<sup>2</sup> transition state display secondary normal alpha H/D KIEs, while reactions with an sp<sup>2</sup> ground state and an sp<sup>3</sup> transition state display secondary inverse KIEs.
Thus, one might think “if iminium formation is rate-limiting, the KIE will be normal, but if alkene addition is rate-limiting, the KIE will be inverse.”
</p>

<p>
Unfortunately this is not true. Instead, all mechanistic possibilities give secondary normal KIEs! I investigated this model system computationally at the wB97X-D/6-31G(d)/SMD(CH<sub>2</sub>Cl<sub>2</sub>) level of theory. Here’s a More O’Ferrall–Jencks plot of the H/D KIE at the iminium position, computed with <a href=https://github.com/ekwan/PyQuiver>PyQuiver</a>
(breaking bond on Y axis, forming bond on X axis): 
</p>

<!--
<figure>
  <img class="centered-img" src="../img/20230103_energy.png" style="width:500px;" />
  <figcaption> 
    Some of the grid points didn't finish, which explains the scattered gaps. I've also excluded anything with energy more than 30 kcal/mol above the ground state.
  </figcaption>
</figure>
-->

<figure>
  <img class="centered-img" src="../img/20230103_kie.png" style="width:550px;" />
  <figcaption> 
    The raw data from PyQuiver is a little fuzzy, so I applied a convolution to smooth the data. Details later on. 
  </figcaption>
</figure>

<p>
Rather than telling us which step is rate-limiting, all the KIE shows us is how much the transition state resembles an iminium ion (bottom right). Structures with long C–Cl bond distances and short C–C bond distances have substantial isotope effects (around 20%), while structures with forming or breaking bonds have smaller isotope effects. 
</p>

<p>
Why is this? Both ionization and addition proceed through iminium-like structures that are substantially sp<sup>2</sup>-hybridized at carbon, irrespective of whether sp<sup>2</sup> character is technically increasing or decreasing in the elementary step. Relative to the starting material, both transition states look like iminium ions and thus lead to large isotope effects. 
</p>

<figure>
  <img class="centered-img" src="../img/20230103_structures2.png" style="width:300px;" />
  <figcaption>
  These both look pretty much like iminium ions.
  </figcaption>
</figure>

<p>
We can also conceptualize this in terms of elementary steps. Taken by itself, alkene addition does lead to an inverse kinetic isotope effect, as seen by the decreasing values as one moves left from the iminium—but an inverse isotope effect relative to the normal equilibrium isotope effect of the iminium. In this system, the equilibrium isotope effect of the iminium is larger than the kinetic isotope effect for alkene addition, and so the combination of these two effects leads to a (smaller) overall normal effect. 
</p>

<p>
(This is the opposite of primary H/D KIEs, where central transition states lead to the largest isotope effects and equilibrium effects are typically small. Here, the isotope effect is mainly a function of hybridization, and so the later the TS, the greater the difference in hybridization and the larger the isotope effect.)
</p>

<p>
In summary, although this experiment seems informative, it’s actually not very useful. It tells you something about the structure of the transition state, but not which step is rate-limiting! In this case, a better experiment would be to measure <sup>13</sup>C KIEs, or an H/D KIE on the nucleophile. 
</p>

<h3>Appendix I: What About S<sub>N</sub>2?</h3>

<p>
On Twitter, <a href="https://twitter.com/LevinChem/status/1610287881993728000">Mark Levin</a> asks about the KIE for the concerted path. I originally meant to include a discussion of this, but then totally forgot to! So let’s fix that.
</p>

<p>
As shown in the graph above, extremely concerted pathways (i.e. going right down the middle of the MOJ plot) will have pretty small isotope effects. These sorts of mechanisms are common where the carbocation would be extremely unstable (methyl iodide) but much less common for stabilized carbocations like we’re discussing here. When oxocarbeniums or iminiums are involved, even “concerted” mechanisms shift towards the bottom right corner: this is the <a href=https://pubs.acs.org/doi/pdf/10.1021/ja506092h>“loose”/“exploded” S<sub>N</sub>2</a> often seen in glycosylation. These pathways will have a modest to large H/D KIE, depending on the exact system and how “exploded” they are (see page 53 of <a href=https://pubs.acs.org/doi/10.1021/jacs.6b10621>this SI</a> for examples)
</p>

<p>
Putting this together, then, what experimental results would be conclusive? A very small KIE would be diagnostic for a “classic” synchronous S<sub>N</sub>2 process, which I consider to be very unlikely here. But medium or large H/D KIEs are consistent with any possibility: S<sub>N</sub>1 with rate-limiting ionization, S<sub>N</sub>1 with rate-limiting nucleophilic attack, or a loose S<sub>N</sub>2. There’s an annulus of different mechanistic possibilities that all give about the same isotope effect, as shown below:
</p>

<figure>
  <img class="centered-img" src="../img/20230103_sn2_plot.png" style="width:470px;" />
  <figcaption> 
  Any TS in the red zone is consistent with a moderately large KIE (say, 15%).
  </figcaption>
</figure>

<p>
To make matters worse, H/D KIEs are pretty tough to simulate quantitatively, because of tunneling, so the annulus isn’t even that precise. That’s why I think this isn’t a very good experiment.
</p>

<h3>Appendix II: Smoothing the KIE Grid</h3>

<p>
The raw KIE values from PyQuiver were pretty noisy, probably because there are small or multiple imaginary frequencies for some of these non-stationary points, so I used convolution to smooth things out a bit. 
</p>

<pre class=code-block>
import numpy as np
from scipy.ndimage import convolve

#### this code block takes a 2d np.ndarray of KIE values 
#### and returns a smoothed np.ndarray with the same dimensions

corner_w = 0.05
edge_w = 0.2
kernel = np.array([
    [corner_w, edge_w, corner_w],
    [edge_w, 1 ,edge_w],
    [corner_w, edge_w, corner_w]
])
kernel = kernel / np.sum(kernel)

smooth_grid = convolve(kie_grid, kernel, mode="nearest")
</pre>

<p>
I haven’t seen this technique used before, but it doesn’t seem unreasonable to me.
</p>
</div><div class='previous-link'><a href='blog_p7.html'>previous page</a></div><div class='next-link'><a href='blog_p9.html'>next page</a></div><br>
  </body>
  <br>
  <footer>
    <a href="mailto:corin.wagen+blog@gmail.com">email</a>
    <a href="https://github.com/corinwagen">github</a>
    <a href="https://twitter.com/CorinWagen">twitter</a>
    <div style="float:right;">
      <a href="/rss.xml">rss</a>
      <a href="https://cwagen.substack.com">substack</a>
    </div>
  </footer>
</html>
