<!DOCTYPE html>
<html>
  <head>
    <title>
      Blog
    </title>
    <meta charset="utf-8">
    <meta name="description" content="">
    <meta name="language" content="en">
    <meta name="viewport" content="width=device-width, initial-scale=1">

    <link rel="stylesheet" href="../../static/style.css">

    <link rel="alternate" type="application/rss+xml" title="RSS feed for the blog" href="/rss.xml">

    <!--google-->
    <script async src="https://www.googletagmanager.com/gtag/js?id=G-MTNZ0ZSG3W"></script>
    <script>
      window.dataLayer = window.dataLayer || [];
      function gtag(){dataLayer.push(arguments);}
      gtag('js', new Date());
      gtag('config', 'G-MTNZ0ZSG3W');
    </script>

  </head>
  <body>
    <ul class="menu-list">
      <li class="menu-item"><a href="index.html" class="menu-link menu-title">Corin Wagen</a></li>
      <li class="menu-item"><a href="index.html#about" class="menu-link">About</a></li>
      <li class="menu-item"><a href="index.html#projects" class="menu-link">Projects</a></li>
      <!--<li class="menu-item"><a href="index.html#past_work" class="menu-link">Past Work</a></li>-->
      <li class="menu-item">
        <a href="blog_p1.html" class="menu-link">Blog</a>
        <a href='archive.html' class="menu-link">(Archive)</a>
      </li>
    </ul>
    <h1 class='blogroll-header'>Blog</h1><div class='previous-link'><a href='blog_p5.html'>previous page</a></div><div class='next-link'><a href='blog_p7.html'>next page</a></div><br><div class='blogroll-container'><h2><a class='blogroll-title' href='../../public/blog/20230417_dimensionality_reduction.html'>Dimensionality Reduction in Cheminformatics</a></h2><i>April 17, 2023</i>
<p>
In many applications, including cheminformatics, it’s common to have datasets that have too many dimensions to analyze conveniently. For instance, <a href=https://www.rdkit.org/UGM/2012/Landrum_RDKit_UGM.Fingerprints.Final.pptx.pdf>chemical fingerprints</a> are typically 2048-length binary vectors, meaning that “chemical space” as encoded by fingerprints is 2048-dimensional. 
</p>

<p>
To more easily handle these complex datasets (and to bypass the <a href=https://en.wikipedia.org/wiki/Curse_of_dimensionality>“curse of dimensionality”</a>), it’s common practice to use a dimensionality reduction algorithm to convert the data to a low-dimensional space. In this post I want to compare and contrast three approaches to dimensionality reduction, and discuss the challenges with low-dimensional embeddings in general.
</p>

<h2>
Dimensionality Reduction Algorithms
</h2>

<p>
There are many approaches to dimensionality reduction, but I’m only going to talk about three here: PCA, tSNE, and UMAP.
</p>

<p>
<a href=https://en.wikipedia.org/wiki/Principal_component_analysis>Principal component analysis</a> (PCA) is perhaps the most famous dimensionality reduction algorithm, and is commonly used in a variety of scientific fields. PCA works by transforming the data into a new set of coordinates such that the first coordinate vector explains the largest amount of the variance, the second coordinate vector the next most variance, and so on and so forth. It’s pretty common for the first 5–20 dimensions to capture &gt;99% of the variance, meaning that the subsequent dimensions can essentially be discarded wholesale. 
</p>

<p>
<a href=https://en.wikipedia.org/wiki/T-distributed_stochastic_neighbor_embedding>tSNE</a> (t-distributed stochastic neighbor embedding) and <a href=https://umap-learn.readthedocs.io/en/latest/how_umap_works.html>UMAP</a> (uniform manifold approximation and projection) are alternative dimensionality reduction approaches, based on much more complex algorithms. To quote Wikipedia: 
</p>

<blockquote>
The t-SNE algorithm comprises two main stages. First, t-SNE constructs a probability distribution over pairs of high-dimensional objects in such a way that similar objects are assigned a higher probability while dissimilar points are assigned a lower probability. Second, t-SNE defines a similar probability distribution over the points in the low-dimensional map, and it minimizes the Kullback–Leibler divergence (KL divergence) between the two distributions with respect to the locations of the points in the map.
</blockquote>

<p>
UMAP, at a high level, works in a very similar way, but uses some fancy topology to construct a “fuzzy simplicial complex” representation of the data in high-dimensional space, and then projects this representation down into a lower dimension (<a href=https://pair-code.github.io/understanding-umap/>more detailed explanation</a>). Practically, UMAP is a lot faster than tSNE, and is becoming the algorithm of choice for most cheminformatics applications. (Although, in fairness, there are <a href=https://arxiv.org/abs/1712.09005>ways to make tSNE faster</a>.)
</p>

<h2>
Data Visualization 
</h2>

<p>
For the purposes of this post, I chose to study Abbie Doyle’s set of <a href=https://pubs.acs.org/doi/10.1021/jacs.1c12203>2683 aryl bromides</a> (obtained from Reaxys, with various filters applied). I used the RDKIT7 fingerprint to generate a 2048-bit encoding of each aryl bromide, computed a distance matrix using Tanimoto/Jaccard distance, and then used each dimensionality reduction technique to generate a 2-dimensional embedding.
</p>

<p>
Let’s look at PCA first:
</p>

<figure>
  <img class="centered-img" src="../img/20230417_pca_plot.png" style="width:400px;" />
  <figcaption> 
  2D plot of aryl bromide chemical space using PCA.
  </figcaption>
</figure>

<p>
PCA generally creates fuzzy-looking blobs, which sometimes show some amount of meaningful structure but don’t really display many sharp boundaries.
</p>

<p>
Now, let’s compare to tSNE:
</p>

<figure>
  <img class="centered-img" src="../img/20230417_tsne_plot.png" style="width:400px;" />
  <figcaption> 
  2D plot of aryl bromide chemical space using tSNE (perplexity 20).
  </figcaption>
</figure>

<p>
tSNE creates “blob-of-blob” plots which show many tight clusters arranged together in some sort of vague pattern. The size and position of the clusters can be tuned by changing the “perplexity” hyperparameter (see <a href=https://stats.stackexchange.com/questions/399868/why-does-larger-perplexity-tend-to-produce-clearer-clusters-in-t-sne>this StackOverflow post</a> for more discussion, and <a href=https://distill.pub/2016/misread-tsne/?s=03>this excellent post</a> for demonstrations of how tSNE can be misleading).
</p>

<p>
What about UMAP?
</p>

<figure>
  <img class="centered-img" src="../img/20230417_umap_plot.png" style="width:400px;" />
  <figcaption> 
  2D plot of aryl bromide chemical space using UMAP (30 neighbors, 0.1 minimum distance).
  </figcaption>
</figure>

<p>
UMAP also creates tight tSNE-like clusters, but UMAP plots generally have a much more variable overall shape—the clusters themselves are tighter and scattered across more space. (These considerations are complicated by the fact that UMAP has multiple tunable hyperparameters, meaning that the exact appearance of the plot is substantially up to the end user.)
</p>

<p>
The debate between tSNE and UMAP is spirited (<a href=https://www.biorxiv.org/content/10.1101/2019.12.19.877522v1.full.pdf>e.g.</a>), but for whatever reason people in chemistry almost exclusively use UMAP. (See, for instance, pretty much every paper I taked about <a href=https://corinwagen.github.io/public/blog/20230118_meta_optimization.html>in this post</a>.)
</p>

<p>
An important thing that I’m not showing here, but which bears mentioning, is that the clusters in all three plots are actually chemically meaningful. For instance, each cluster in the tSNE plot generally corresponds to a different functional group: carboxylic acids, alkynes, etc. So the graphs do in some real sense correspond to the intuition we have about molecular similarity, which is good! (You can use <a href=https://github.com/wjm41/molplotly>molplotly</a> to visualize these plots very easily.)
</p>

<h2>
Distance Preservation
</h2>

<p>
How well are distances from the high-dimensional space preserved in the 2D embedding? Obviously the distances won’t all be the same, but ideally the mapping would be monotonic: if distance A is greater than distance B in the high-dimensional space, we would like distance A to also be greater than distance B in the low-dimensional space. 
</p>

<p>
We can measure this with <a href=https://en.wikipedia.org/wiki/Spearman%27s_rank_correlation_coefficient>Spearman correlation</a>, which is like a Pearson correlation (AKA “r-squared”) but without the assumption of linearity. A Spearman correlation coefficient of 1 indicates a perfect monotonic relationship, while a coefficient of 0 indicates no relationship. Let’s plot the pairwise distances from each embedding against the true distances and compare the Spearman coefficients:
</p>

<figure>
  <img class="centered-img" src="../img/20230417_spearman_plots.png" style="width:300px;" />
  <figcaption> 
  Comparison of distances in high-dimensional space against distances in embedding space, and associated Spearman coefficients. (Only one in every hundred points is plotted, but all points are used for the Spearman coefficient calculation.)
  </figcaption>
</figure>

<p>
In each case, the trend is in the right direction (i.e. increased distance in high-dimensional space is correlated with increased distance in low-dimensional space), but the relationship is far from monotonic. It’s clear that there will be plenty of cases where two points will be close in low-dimensional space and far in high-dimensional space. 
</p>

<p>
Does this mean that UMAP, tSNE, and PCA are all failing? To understand this better, let’s plot a histogram of all the distances in each space:
</p>

<figure>
  <img class="centered-img" src="../img/20230417_dist_hist.png" style="width:400px;" />
  <figcaption> 
  Histogram of all distances in each space. Distances have been scaled to the range [0,1] to match distances obtained with the Jaccard metric.
  </figcaption>
</figure>

<p>
We can see that the 2048-dimensional space has a very distinct histogram. Most of the compounds are pretty different from one another, and—crucially—most of the distances are about the same (0.8 or so). In chemical terms, this means that most of the fingerprints share a few epitopes in common, but otherwise are substantially different, which is unsurprising since fingerprints in general are quite sparse. 
</p>

<p>
Unfortunately, “lots of equidistant points” is an extremely tough pattern to recapitulate in a low-dimensional space. We can see why with a toy example: in 2D space, we can only have 3 equidistant points (an equilateral triangle), and in 3D space, we can only have 4 equidistant points (a tetrahedron). More generally, if we want <i>N</i> equidistant points, we need to be in <b>R</b><sup><i>N</i>-1</sup> (<i>N</i>-1 dimensional Euclidean space). We can <a href=https://en.wikipedia.org/wiki/Johnson%E2%80%93Lindenstrauss_lemma>relax this requirement</a> a little bit if we’re willing to accept approximate equidistance, but the general principle still holds: it’s hard to recapitulate lots of equidistant points in a low-dimensional space. 
</p>

<p>
As expected, then, we can see that the histogram of each of our algorithms looks very different from the ideal distance histogram.
</p>

<h2>
Local Structure
</h2>

<p>
Both tSNE and UMAP take the nearest neighbors of each point explicitly into account, and claim to preserve the local structure of the points as much as possible. To put these claims to the test, I looked at the closest 30 neighbors of each point in high-dimensional space, and then checked how many of those neighbors made it into the closest 30 neighbors in low-dimensional space. 
</p>

<figure>
  <img class="centered-img" src="../img/20230417_neighbor_hist.png" style="width:400px;" />
  <figcaption> 
  Histogram of how many of the closest 30 neighbors of each point are recapitulated after dimensionality reduction.
  </figcaption>
</figure>

<p>
We can see that PCA only preserves about 30–40% of each point’s neighbors, whereas PCA and UMAP generally preserve 60% of the neighbors: not perfect, but much better.
</p>

<p>
I chose to look at 30 neighbors somewhat arbitrarily: what happens if we change this number?
</p>

<figure>
  <img class="centered-img" src="../img/20230417_neighborhood_scan.png" style="width:400px;" />
  <figcaption> 
  The percent of neighbors recapitulated correctly, as neighborhood size increases.
  </figcaption>
</figure>

<p>
We can see that UMAP and tSNE both preserve about 60% of the neighbors across a wide range of neighborhood sizes, while PCA gets better as we zoom out more. (At the limit where we consider all 2683 points as neighbors, every method will trivially achieve perfect accuracy.) tSNE does much better than UMAP for small neighborhoods; I’m not sure why!
</p>

<p>
Another way to think about this is in terms of the <a href=https://scikit-learn.org/stable/auto_examples/model_selection/plot_precision_recall.html>precision–recall tradeoff</a>. In classification, “precision” refers to a classifier’s ability to avoid false positives, while “recall” refers to a classifier’s ability to avoid false negatives. What does this mean in the context of embedding? 
</p>

<p>
Imagine looking at all points in the neighborhood of our central point in high-dimensional space, and then comparing to the points within a certain radius of our point in low-dimensional space. As we increase the radius, we expect to see more of the correct neighbor points in low-dimensional space, but we also expect to see more “incorrect neighbors” that aren’t really there in the high-dimensional space. (<a href=https://jmlr.org/papers/volume11/venna10a/venna10a.pdf>This paper</a> discusses these issues nicely, as does <a href=https://coursepages2.tuni.fi/mttts17/wp-content/uploads/sites/136/2020/03/drv_2020_lecture_7.pdf>this presentation</a>.)
</p>

<p>
So low radii lead to high precision (most of the points are really neighbors) but low recall (we’re not finding most of the neighbors), while high radii lead to low precision and high recall. We can thus study the performance of our embedding by graphing the precision–recall curve for various neighborhood sizes. The better the embedding, the closer the curve will come to the top right:
</p>

<figure>
  <img class="centered-img" src="../img/20230417_precision_recall.png" style="width:400px;" />
  <figcaption> 
  Precision–recall tradeoff for all three methods.
  </figcaption>
</figure>

<p>
We can see that tSNE does better in the high precision/low recall area of the curve (as we saw in the previous graph), but otherwise tSNE and UMAP are quite comparable. In contrast, PCA is just abysmal.
</p>

<p>
The big conclusion of this section is that, if you’re doing something that depends on the local structure of the data, you should avoid PCA. 
</p>

<h2>
Do Higher Dimensions Help Things? 
</h2>

<p>
Since the root of our issues here is trying to represent a 2048-dimensional distance matrix in 2 dimensions, one might wonder if we could do better by expanding to 3, 4, or more dimensions. This would make visualization tricky, but might still be suitable for other operations (like clustering). 
</p>

<p>
tSNE gets very, very slow in higher dimensions, so I focused on PCA and UMAP for this study. I started out by comparing the Spearman correlation for PCA and UMAP up to 20 dimensions:
</p>

<figure>
  <img class="centered-img" src="../img/20230417_spearman_dim.png" style="width:400px;" />
  <figcaption> 
  Precision–recall tradeoff for all three methods.
  </figcaption>
</figure>

<p>
Surprisingly, UMAP doesn’t seem to get any better in high dimensions, but PCA does. (Changing the number of neighbors didn’t help UMAP at all.)
</p>

<p>
How do our other metrics look with high-dimensional PCA?
</p>

<figure>
  <img class="centered-img" src="../img/20230417_10dim_dist_hist.png" style="width:400px;" />
  <figcaption> 
  Distance histogram for PCA with 2–10 dimensions.
  </figcaption>
</figure>

<p>
As we increase the number of dimensions, the distance histogram starts to approach the correct distribution. 
</p>

<figure>
  <img class="centered-img" src="../img/20230417_10dim_neighbor_hist.png" style="width:400px;" />
  <figcaption> 
  Neighbor histogram for PCA with 2–10 dimensions.
  </figcaption>
</figure>

<p>
We also start to do a better job capturing the local structure of the graph, although we’re still not as good as tSNE or UMAP even at 10 dimensions. 
</p>

<figure>
  <img class="centered-img" src="../img/20230417_10dim_precision_recall.png" style="width:400px;" />
  <figcaption> 
  Precision–recall curve for PCA with 2–10 dimensions.
  </figcaption>
</figure>

<p>
And our precision–recall graph is still pretty dismal when compared to tSNE or UMAP. So, it seems like if distances are what matters, then high-dimensional PCA is an appealing choice—but if local structure is what matters, tSNE or UMAP is still superior.
</p>

<h2>
Conclusions
</h2>

<p>
My big takeaway from all of this is: dimensionality reduction is a lossy process, and one where you always have to make tradeoffs. You’re fundamentally throwing away information, and that always has a cost: there’s no such thing as a free lunch. As such, if you don’t have to perform dimensionality reduction, then my inclination would be to avoid it. (People in single-cell genomics seem to have come to <a href=https://www.biorxiv.org/content/10.1101/2021.08.25.457696v4.full.pdf>a similar conclusion</a>.)
</p>

<p>
If you really need your data to be in a low-dimensional space (e.g. for plotting), then keep in mind what you’re trying to study! PCA seems to do a slightly better job with distances (although I’m sure there are more sophisticated strategies for distance-preserving dimensionality reduction), while tSNE and UMAP seem to do much, much better with local structure. 
</p>

<i>
Thanks to Michael Tartre for helpful conversations, and the students in Carnegie Mellon’s “Digital Molecular Design Studio” class for their thought-provoking questions on these topics.  
</i>

</div><div class='blogroll-container'><h2><a class='blogroll-title' href='../../public/blog/20230413_new_ways.html'>New Ways To Read The Blog: RSS and Substack</a></h2><i>April 13, 2023</i>
<p>
<i>
(This is more of a housekeeping post than an actual post with content; apologies.)
</i>
</p>

<p>
Up until now, my blogging strategy has been to write new posts about once a week and publicize them on Twitter, which works great for people who are on Twitter but (obviously) fails for people who aren’t on Twitter. I’m frequently asked if there are non-Twitter ways to subscribe to the blog updates: given that I myself don’t love relying on Twitter to bring me content, and that Twitter itself feels increasingly dicey, I feel bad saying no every time.
</p>

<p>
I’m happy to announce that there are now two additional ways to read the blog: RSS and Substack. 
</p>

<h2>
RSS
</h2>

<p>
RSS is a lovely way to get updates from sites, which is sadly limited by the fact that nobody uses it anymore. (Half the people I talk to these days don’t even know what it is.) You can use an RSS aggregator like <a href=https://feedly.com/>Feedly</a>, and simply subscribe to various sites, so that they’ll dependably show up in your feed. This is the main way I get <a href=https://corinwagen.github.io/public/blog/20230329_literature.html>journal updates</a> and my news. 
</p>

<p>
So, if you like using RSS, you can simply search “corinwagen.github.io” in Feedly, and the blog will come up:
</p>

<figure>
  <img class="centered-img" src="../img/20230413_feedly.jpg" style="width:300px;" />
  <figcaption> 
  What it looks like on Feedly. The Twitter preview images sadly don't display.
  </figcaption>
</figure>

<h2>
Substack
</h2>

<p>
Substack is a platform that helps people write and manage newsletters. It essentially solves the problem of “how do I create an email list”/“how do I manage subscriptions” for people who would rather not take care of hosting a web service and handling payments themselves, like me.
</p>

<p>
I initially didn’t want to use Substack because (1) I wanted the blog to be part of my website, (2) I liked being able to control every aspect of the design, and (3) I wasn’t sure if anyone would read the blog, and there’s nothing sadder than an empty Substack. As things stand, (3) is a non-issue, so the question is whether the added convenience of Substack outweighs my own personal design and website preferences.
I suspect that it may, so I’ve capitulated and copied all existing posts over to <a href=https://cwagen.substack.com>my new Substack</a>. (There are a few formatting issues in old posts, but otherwise things copied pretty well.)
</p>

<p>
For now, I plan to continue posting everything on the blog, and manually copying each post over to Substack (I write in plain HTML so this is not too hard). If Substack ends up totally outperforming the blog in terms of views, then I’ll probably switch to Substack entirely for blogging and just leave my website up as a sort of virtual CV. 
</p>

<p>
(I have no plans to enable subscriptions at this point; that being said, if for some bizarre reason there’s sufficient demand I’ll probably try to think of something to reward subscribers.)
</p>

<p>
If you’d like to receive updates on Substack, you can subscribe below: 
</p>

<br>
<iframe src="https://cwagen.substack.com/embed" width="480" height="320" style="display:block; border:1px solid #EEE; background:white; margin:auto;" frameborder="0" scrolling="no"></iframe>

</div><div class='blogroll-container'><h2><a class='blogroll-title' href='../../public/blog/20230411_newcomers.html'>Why New Ventures Are So Important</a></h2><i>April 11, 2023</i>
<p>
This Easter week, I’ve been thinking about why new ventures are so important. Whether in private industry, where startups are the most consistent source of innovative ideas, or in academia, where new assistant professors are hired each year, newcomers are often the most consistent source of innovation. Why is this?
</p>

<p>
One explanation is the <a href=https://www.nber.org/system/files/chapters/c2144/c2144.pdf>Arrow replacement effect</a> (named after <a href=https://en.wikipedia.org/wiki/Kenneth_Arrow>Kenneth Arrow</a>), which states that “preinvention monopoly power acts as a strong disincentive to further innovation.” Arrow’s argument goes like this: suppose there’s an organization that is earning profit <i>P<sub>old</sub></i>, and there is some innovation that will increase profit to <i>P<sub>new</sub></i> (<i>P<sub>new</sub></i> &gt; <i>P<sub>old</sub></i>). If the existing organization pursues the innovation, their profits will thus increase by <i>∆P</i> := <i>P<sub>new</sub></i> - <i>P<sub>old</sub></i>. But a new organization will see its profits increase by <i>P<sub>new</sub></i>: since the startup has no existing profit to replace, the rewards to innovation are higher. Thus innovation is more appealing for those without any economic stake in the status quo.<sup><a href="#fn1">1</a></sup>
</p>

<p>
We can see this play out today in the dynamic between Google and OpenAI/Microsoft: Google already has a virtual monopoly in search, and so is hesitant to replace what they have, whereas Microsoft has already been losing in search and so is eager to replace Bing with <s>Sydney</s> an AI-powered alternative. (It’s to Apple’s credit that they so eagerly pursued the iPhone when it meant effectively destroying the iPod, one of their top money-makers.<sup><a href="#fn2">2</a></sup>)
</p>

<p>
One can also see this scenario in academia—plenty of established labs have programs built up around studying specific systems, and are thus disincentivized to study areas which might obviate projects they’ve spent decades working on. For instance, labs dedicated to “conventional” synthetic methodology might be slower to turn to biocatalysis than a new assistant professor with nothing to lose; labs that have spent decades studying protein folding might be slower to turn to AlphaFold than they ought to.
</p>

<p>
Another reason is that new entrants often have an advantage in understanding the status quo. In <i>The Art of Doing Science and Engineering</i> (book review coming, eventually), computing legend <a href=https://en.wikipedia.org/wiki/Richard_Hamming>Richard Hamming</a> discusses how there’s often a disadvantage to being a pioneer in a field. Hamming’s argument, essentially, is that those who’ve had to invent something new never understand it as intuitively as those who have simply learned to take it for granted:
</p>

<blockquote>
The reason this happens is that the creators have to fight through so many dark difficulties, and wade through so much misunderstanding and confusion, they cannot see the light as others can, now the door is open and the path made easy…. in time the same will probably be true of you.
</blockquote>

<p>
In Hamming’s view, it’s the latecomers to a field who can see more clearly the new possibilities opened by various innovations, and take the next steps towards previously unimaginable frontiers. There’s a deep irony in this: the very act of inventing something new makes you less able to see the innovations enabled by your own work. The process of invention thus acts like a relay race, where newer generations continually take the baton and push things forward before in turn dropping back.
</p>

<p>
I’ve heard these ideas discussed in terms of naïvete before—the idea being that innovation requires a sort of “beginner’s luck,” a blind optimism about what’s possible that the experienced lack—but I think that’s wrong. A belief in naïvete as the key driver of innovation implies that excessive knowledge is detrimental: that it’s possible to “know too much” and cripple oneself. If anything, the opposite is true in my experience. The most creative and productive people I’ve met are those with an utter mastery of the knowledge in their domain.
</p>

<p>
Hamming’s proposal, which is more cognitive/subconscious, is thus complementary to the more calculated logic of the Arrow replacement theorem: existing organizations are both less incentivized to innovate and less able to see potential innovations. These ideas should be encouraging to anyone at the beginning of their career: you are uniquely poised to discover and exploit new opportunities! So consider this an exhortation to go out and do so now (rather than waiting until you are older and more secure in your field).
</p>

<i>Credit to <a href="https://marginalrevolution.com/marginalrevolution/2023/04/the-arrow-replacement-effect-and-the-dynamics-of-us-inventors.html?utm_source=feedly&utm_medium=rss&utm_campaign=the-arrow-replacement-effect-and-the-dynamics-of-us-inventors">Alex Tabarrok</a> for introducing me to the Arrow replacement effect, and ChatGPT for some edits.</i>

<h3>Footnotes</h3>
<ol class=footnotes>
  <li id="fn1">
  This is a bit of a cartoonish depiction of the Arrow replacement theorem—the original paper (linked above) is quite readable, and performs a more sophisticated analysis. See the heading “Competition, Monopoly, and the Incentive to Innovate” on page 12 of the PDF (journal page 619).
  </li>
  <li id="fn2">
  Tony Fadell discusses this in <a href=https://www.amazon.com/Build-Unorthodox-Guide-Making-Things/dp/0063046067><i>Build</i></a>: suffice it to say this was not an internally popular decision at Apple.
  </li>
</ol>
</div><div class='blogroll-container'><h2><a class='blogroll-title' href='../../public/blog/20230403_industry.html'>Industry Research Seems Underrated</a></h2><i>April 3, 2023</i>
<p>
While scientific companies frequently publish their research in academic journals, it seems broadly true that publication is not incentivized for companies the same way it is for academic groups. Professors need publications to get tenure, graduate students need publications to graduate, postdocs need publications to get jobs, and research groups need publications to win grants. So the incentives of everyone in the academic system are aligned towards publishing papers, and lots of papers get published.
</p>

<p>
In contrast, the success or failure of a private company is—to a first approximation—unrelated to its publication record. Indeed, publication might even be harmful for companies, insofar as time spent preparing manuscripts and acquiring data <a href=https://corinwagen.github.io/public/blog/20221214_against_carbon_nmr.html>only needed for publication</a> is time that could be spent on more mission-critical activities.
</p>

<p>
That’s why I generally believe industry publications, especially those where no academic co-authors are involved, are underrated, and are probably better than the journal they’re in might indicate. Getting a publication into a prestigious journal like <i>Science</i> or <i>Nature</i> is pretty random, requires a lot of effort, and frequently has a slow turnaround time, whereas lower-tier journals are likely to accept your work, and typically review and publish papers much, much faster. (In particular, ACS is <a href=https://scholarlykitchen.sspnet.org/2022/11/08/guest-post-publishing-fast-and-slow-a-review-of-publishing-speed-in-the-last-decade/>among the fastest of all scientific publishers</a>, and is generally a pleasure to work with.)
</p>

<p>
The above reflections were prompted by reading <a href=https://pubs.acs.org/doi/full/10.1021/acs.jmedchem.0c00452>an absolute gem of a paper</a> in <i>J. Med. Chem.</i>, a collaboration between X-Chem, ZebiAI, and Google Research. The paper is entitled “Machine Learning on DNA-Encoded Libraries: A New Paradigm for Hit Finding” and describes how data from DNA-encoded libraries (DELs) can be used to train ML models to predict commercially available compounds with activity against a given target. This is a really, really big deal. As the authors put it in their conclusion:
</p>

<blockquote>
[Our approach] avoids the time-consuming and expensive process of building new chemical matter into a DEL library and performing new selections or incorporating new molecules into a HTS screening library. This ability to consider compounds outside of the DEL is the biggest advantage of our approach; notably, this approach can be used at a fraction of the cost of a traditional DEL screening follow-up, driven primarily by the large difference in synthesis cost.
</blockquote>

<p>
Now, the precise impact of this discovery will of course be determined in the years to come; Derek Lowe raises some fair concerns <a href=https://www.science.org/content/blog-post/machine-learning-top-dna-encoded-libraries>on his blog</a>, pointing out that the targets chosen are relatively easy to drug, and so probably wouldn’t be the subject of a high-tech DEL screen anyway, and it’s entirely possible that there will be other unforeseen complications with this technology that are only revealed in the context of a real-world discovery pipeline. (Given that Relay <a href=https://www.biopharmadive.com/news/relay-acquire-zebiai-ai-drug-discovery/598550/>acquired ZebiAI</a> for $85M in 2021 essentially on the strength of this paper alone, I’m guessing plenty of real-world testing is already underway.)
</p>

<p>
The point I want to make is that if this paper had come from an academic group, I would be very, very surprised to see it in <i>J. Med Chem</i>. This project has everything that one expects in a <i>Science</i> paper: a flashy new piece of technology, a problem that’s understandable to a broad audience, clear clincal relevance, even a domain arbitrage angle. Yet this paper is not in <i>Science</i>, nor <i>ACS Central Science</i>, nor even <i>JACS</i>, but in <i>J. Med. Chem.</i>, a journal I don’t even read regularly.
</p>

<p>
My conclusions from this are (1) to remember that not everyone is incentivized to market their own findings as strongly as academics are and (2) to try and look out for less-hyped industry results that I might neglect otherwise.
</p>

</div><div class='blogroll-container'><h2><a class='blogroll-title' href='../../public/blog/20230329_literature.html'>You Should Read The Literature More</a></h2><i>March 29, 2023</i>
<p>
If you are a scientist, odds are you should be reading the literature more. This might not be true in every case—one can certainly imagine someone who reads the literature too much and never does any actual work—but as a heuristic, my experience has been that most people would benefit from reading more than they do, and often much more. Despite the somewhat aggressive title, my hope in writing this is to encourage you to read the literature more: to make you excited about reading the literature, not to guilt you into it or provoke anxiety.
<p>

<h2>
Why You Should Read The Literature More
</h2>

<p>
You should read the literature because you are a scientist, and your business is ideas. The literature is the vast, messy, primal chaos that contains centuries of scientific ideas. If you are an ideas worker, this is your raw material—this is what you work with. Not reading the literature as an ideas worker is like not going to new restaurants as a new chef, or not looking at other people’s art as an artist, or not listening to music as a composer. Maybe the rare person has an internal creativity so deep that they don’t need any external sources of inspiration, but I’m not sure I know anyone like that.
</p>

<p>
If you buy the concept of <a href=https://corinwagen.github.io/public/blog/20230320_domain_arbitrage.html>“domain arbitrage”</a> I outlined last week, then reading the literature becomes doubly important for up-and-coming arbitrageurs. Not only do you need to stay on top of research in your own field, but you also need to keep an eye on other fields, to look for unexpected connections. It was only after months of reading physical chemistry papers about various IR spectroscopy techniques, with no direct goal in mind, that I realized I could use <i>in situ</i> IR to <a href=https://pubs.acs.org/doi/10.1021/acs.orglett.2c03622>pin down the structure of ethereal HCl</a>; simply reading organic chemistry papers would not have given me that insight.
</p>

<h2>
How You Can Read The Literature More
</h2>

<p>
If you don’t read the literature at all—like me, when I started undergrad—then you should start small. I usually recommend <i>JACS</i> to chemists. Just try to read every paper in your subfield in <i>JACS</i> for a few months; I began by trying to read every organic chemistry paper in <i>JACS</i>. At the beginning, probably only 10–20% will make sense. But if you push through and keep trying to make sense of things, eventually it will get easier. You’ll start to see the same experiments repeated, understand the structure of different types of papers, and even recognize certain turns of phrase. (This happened to me after about a year and a half of reading the literature.)
</p>

<p>
Reading more papers makes you a faster reader. Here’s <a href=https://marginalrevolution.com/marginalrevolution/2006/12/how_to_read_fas.html>Tyler Cowen</a> on how he reads so quickly (not papers specifically, but still applicable):
</p>

<blockquote>
The best way to read quickly is to read lots. And lots. And to have started a long time ago. Then maybe you know what is coming in the current book. Reading quickly is often, in a margin-relevant way, close to not reading much at all.
<br><br>
Note that when you add up the time costs of reading lots, quick readers don’t consume information as efficiently as you might think. They’ve chosen a path with high upfront costs and low marginal costs. "It took me 44 years to read this book" is not a bad answer to many questions about reading speed.
</blockquote>

<p>
All of Tyler’s advice applies doubly to scientific writing, which is often jargon-filled and ordered in arcane ways. After 7ish years of reading the scientific literature, I can “skim” a <i>JACS</i> paper pretty quickly and determine what, if anything, is likely to be novel or interesting to me, which makes staying on top of the literature much easier than it used to be.
</p>

<p>
Once you are good with a single journal, you can expand to multiple journals. A good starting set for organic chemistry is <i>JACS</i>, <i>Science</i>, <i>Nature</i>, <i>Nature Chemistry</i>, and <i>Angewandte</i>. If you already know how to read papers quickly, it will not be very hard to read more and more papers. But expanding to new journals brings challenges: how do you keep up with all of them at once? Lots of people use an RSS feed to aggregate different journals—I use <a href=https://feedly.com/>Feedly</a>, as do several of my coworkers. (You can also get this blog on Feedly!)
</p>

<p>
I typically check Feedly many times a day on my phone; I can look at the TOC graphic, the abstract, and the title, and then if I like how the paper looks I’ll email it to myself. Every other day or so, I sit down at my computer with a cup of coffee and read through the papers I’ve emailed to myself. This is separate from my “pursuing ideas from my research”/”doing a literature dive for group meeting” way of reading the literature—this is just to keep up with all the cool stuff that I wouldn’t otherwise hear about.
</p>

<figure>
  <img class="centered-img" src="../img/20230329_email.png" style="width:550px;" />
  <figcaption> 
  “Inbox Zero” often proves elusive.
  </figcaption>
</figure>

<p>
(I also use Google Scholar Alerts to email me when new labs publish results—I have probably 20-30 labs set up this way, just to make sure I don’t miss results that might be important just because they’re not in a high-profile journal.)
</p>

<p>
Keeping track of papers you actually like and want to remember is another challenge. For the past two years, I’ve put the URLs into a Google Sheet, along with a one-sentence summary of the paper, which helps me look through my “most liked” papers when I want to find something. Sadly, I didn’t do this earlier, so I’m often tormented by papers I dimly remember but can no longer locate.
</p>

<h2>
What Literature You Should Read
</h2>

<p>
This obviously depends on what you’re doing, but I tend to think about literature results in three categories:
</p>

<ol>
<li>Things every scientist should know about</li>
<li>Things I am supposed to be an expert on</li>
<li>Things I’m not supposed to be an expert on, but would still like to know about</li>
</ol>

<p>
Category 1 basically covers the highest profile results (<i>Science</i> and <i>Nature</i>), and these days Twitter makes that pretty easy.
</p>

<p>
Category 2 covers things “in-field” or directly related to my projects—anything it would be somewhat embarrassing not to know about. For me, this means <i>JACS</i>, <i>Angewandte</i>, <i>ACS Catalysis</i>, <i>Org. Lett.</i>, <i>OPRD</i>, <i>Organometallics</i>, <i>J. Org. Chem.</i>, and <i>Chem. Sci.</i> (I also follow <i>Chem. Rev.</i> and <i>Chem. Soc. Rev.</i>, because review articles are nice.)
</p>

<p>
Category 3 covers things that I am excited to learn about. Right now, that’s <i>JCTC</i> and <i>J. Phys. Chem. A–C</i>. In the past, that’s included <i>ACS Chem. Bio.</i>, <i>Nature Chem. Bio.</i>, and <i>Inorganic Chemistry</i>. (Writing this piece made me realize I should follow <i>JCIM</i> and <i>J. Chem. Phys.</i>, so I just added them to Feedly.)
</p>

<h2>
Conclusion
</h2>

<p>
Reading the literature is—in the short term—pointless, sometimes frustrating, and just a waste of time. It’s rare that the article you read today will lead to an insight on the problem you’re currently facing! But the gains to knowledge compound over time, so spending time reading the literature today will make you a much better scientist in the long run.
</p>

<i>
Thanks to Ari Wagen and Joe Gair for reading drafts of this post.
</i>

</div><div class='previous-link'><a href='blog_p5.html'>previous page</a></div><div class='next-link'><a href='blog_p7.html'>next page</a></div><br>
  </body>
  <br>
  <footer>
    <a href="mailto:corin.wagen+blog@gmail.com">email</a>
    <a href="https://github.com/corinwagen">github</a>
    <a href="https://twitter.com/CorinWagen">twitter</a>
    <div style="float:right;">
      <a href="/rss.xml">rss</a>
      <a href="https://cwagen.substack.com">substack</a>
    </div>
  </footer>
</html>
